src/configuration/preprocessors/api/actions.coffee

```diff
@@ -21,13 +22,38 @@ module.exports = (description) ->
   hasDependency = (signature) ->
     if signature.request || signature.response then true else false
 
+  dependencies = ({request, response}) ->
+    result = []
+    if request
+      result.push "#{capitalize(request)}Resource"
+      result.push "#{capitalize(request)}Model"
+    if response
+      result.push "#{capitalize(response)}Resource"
+      result.push "#{capitalize(response)}Model"
+    result
+
   for r, resource of resources
     methods = ["options"]
     for a, action of resource.actions
+      action.method = capitalize action.method
       methods.push action.method
-      resources[r]["actions"][a].camelName = makeCamelName r, action
-      resources[r]["actions"][a].dashName = makeDashName r, action
-      resources[r]["actions"][a]["signature"].dependent = hasDependency action.signature
+      action.camelName = makeCamelName r, action
```

Comment:
This all seems like a giant step in the right direction. I wanted to see how far I could get using just the API description and the template, but the result is almost unmaintainable. The trick is figuring out what these things are (ex: gateway method name) but I think this does a pretty job of doing that.